src/channel_layout.c

```diff
@@ -305,357 +307,226 @@ static struct SoundIoChannelLayout builtin_channel_layouts[] = {
+    names[SoundIoChannelIdHearingImpaired][1] = NULL;
+    names[SoundIoChannelIdHearingImpaired][2] = NULL;
+    names[SoundIoChannelIdNarration][0] = "Narration";
+    names[SoundIoChannelIdNarration][1] = NULL;
+    names[SoundIoChannelIdNarration][2] = NULL;
+    names[SoundIoChannelIdHaptic][0] = "Haptic";
+    names[SoundIoChannelIdHaptic][1] = NULL;
+    names[SoundIoChannelIdHaptic][2] = NULL;
+    names[SoundIoChannelIdDialogCentricMix][0] = "Dialog Centric Mix";
+    names[SoundIoChannelIdDialogCentricMix][1] = NULL;
+    names[SoundIoChannelIdDialogCentricMix][2] = NULL;
+    names[SoundIoChannelIdAux][0] = "Aux";
+    names[SoundIoChannelIdAux][1] = NULL;
+    names[SoundIoChannelIdAux][2] = NULL;
+    names[SoundIoChannelIdAux0][0] = "Aux 0";
+    names[SoundIoChannelIdAux0][1] = NULL;
+    names[SoundIoChannelIdAux0][2] = NULL;
+    names[SoundIoChannelIdAux1][0] = "Aux 1";
+    names[SoundIoChannelIdAux1][1] = NULL;
+    names[SoundIoChannelIdAux1][2] = NULL;
+    names[SoundIoChannelIdAux2][0] = "Aux 2";
+    names[SoundIoChannelIdAux2][1] = NULL;
+    names[SoundIoChannelIdAux2][2] = NULL;
+    names[SoundIoChannelIdAux3][0] = "Aux 3";
+    names[SoundIoChannelIdAux3][1] = NULL;
+    names[SoundIoChannelIdAux3][2] = NULL;
+    names[SoundIoChannelIdAux4][0] = "Aux 4";
+    names[SoundIoChannelIdAux4][1] = NULL;
+    names[SoundIoChannelIdAux4][2] = NULL;
+    names[SoundIoChannelIdAux5][0] = "Aux 5";
+    names[SoundIoChannelIdAux5][1] = NULL;
+    names[SoundIoChannelIdAux5][2] = NULL;
+    names[SoundIoChannelIdAux6][0] = "Aux 6";
+    names[SoundIoChannelIdAux6][1] = NULL;
+    names[SoundIoChannelIdAux6][2] = NULL;
+    names[SoundIoChannelIdAux7][0] = "Aux 7";
+    names[SoundIoChannelIdAux7][1] = NULL;
+    names[SoundIoChannelIdAux7][2] = NULL;
+    names[SoundIoChannelIdAux8][0] = "Aux 8";
+    names[SoundIoChannelIdAux8][1] = NULL;
+    names[SoundIoChannelIdAux8][2] = NULL;
+    names[SoundIoChannelIdAux9][0] = "Aux 9";
+    names[SoundIoChannelIdAux9][1] = NULL;
+    names[SoundIoChannelIdAux9][2] = NULL;
+    names[SoundIoChannelIdAux10][0] = "Aux 10";
+    names[SoundIoChannelIdAux10][1] = NULL;
+    names[SoundIoChannelIdAux10][2] = NULL;
+    names[SoundIoChannelIdAux11][0] = "Aux 11";
+    names[SoundIoChannelIdAux11][1] = NULL;
+    names[SoundIoChannelIdAux11][2] = NULL;
+    names[SoundIoChannelIdAux12][0] = "Aux 12";
+    names[SoundIoChannelIdAux12][1] = NULL;
+    names[SoundIoChannelIdAux12][2] = NULL;
+    names[SoundIoChannelIdAux13][0] = "Aux 13";
+    names[SoundIoChannelIdAux13][1] = NULL;
+    names[SoundIoChannelIdAux13][2] = NULL;
+    names[SoundIoChannelIdAux14][0] = "Aux 14";
+    names[SoundIoChannelIdAux14][1] = NULL;
+    names[SoundIoChannelIdAux14][2] = NULL;
+    names[SoundIoChannelIdAux15][0] = "Aux 15";
+    names[SoundIoChannelIdAux15][1] = NULL;
+    names[SoundIoChannelIdAux15][2] = NULL;
+    return &names[0];
 };
+static channel_names_t *channel_names = make_channel_names();
```

Comment:
Same thing here. Let's make this not require running code at runtime. Let's just carefully line up the array initializer with  the enum values.